### ISSUE TITLE
[Extension manager GUI]: getting error "... is not a valid npm package"

### ISSUE THREAD

**Issue by Zireael** [jupyterlab/jupyterlab#7531]
Hi, I build a docker image for Jupyterlab and then install a bunch of extensions through conda/pip in dockerfile.
When I run the docker image and go to [Extension Manager] tab, I get the following error after timeout:

> Error communicating with server extension. Consult the documentation for how to ensure that it is enabled.
Reason given:
Error: 500 (Internal Server Error)

Console log: 
> [E 23:57:52.995 LabApp] 500 GET /lab/api/extensions?1573862235795 (172.23.0.1) 36213.41ms referer=http://localhost:8888/lab?
> jupyterlab    | [E 23:57:52.995 LabApp] Uncaught exception GET /lab/api/extensions?1573862235875 (172.23.0.1)
> jupyterlab    |     HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/lab/api/extensions?1573862235875', version='HTTP/1.1', remote_ip='172.23.0.1')
> jupyterlab    |     Traceback (most recent call last):
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/web.py", line 1699, in _execute
> jupyterlab    |         result = await result
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/web.py", line 1699, in _execute
> jupyterlab    |         result = await result
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 742, in run
> jupyterlab    |         yielded = self.gen.throw(*exc_info)  # type: ignore
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/jupyterlab/handlers/extension_manager_handler.py",
> line 222, in get
> jupyterlab    |         extensions = yield self.manager.list_extensions()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 735, in run
> jupyterlab    |         value = future.result()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 742, in run
> jupyterlab    |         yielded = self.gen.throw(*exc_info)  # type: ignore
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/jupyterlab/handlers/extension_manager_handler.py",
> line 222, in get
> jupyterlab    |         extensions = yield self.manager.list_extensions()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 735, in run
> jupyterlab    |         value = future.result()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 742, in run
> jupyterlab    |         yielded = self.gen.throw(*exc_info)  # type: ignore
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/jupyterlab/handlers/extension_manager_handler.py",
> line 92, in list_extensions
> jupyterlab    |         pkg_info = yield self._get_pkg_info(name, data)
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 735, in run
> jupyterlab    |         value = future.result()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 742, in run
> jupyterlab    |         yielded = self.gen.throw(*exc_info)  # type: ignore
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/jupyterlab/handlers/extension_manager_handler.py",
> line 92, in list_extensions
> jupyterlab    |         pkg_info = yield self._get_pkg_info(name, data)
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 735, in run
> jupyterlab    |         value = future.result()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 742, in run
> jupyterlab    |         yielded = self.gen.throw(*exc_info)  # type: ignore
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/jupyterlab/handlers/extension_manager_handler.py",
> line 161, in _get_pkg_info
> jupyterlab    |         outdated = yield self._get_outdated()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 735, in run
> jupyterlab    |         value = future.result()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 742, in run
> jupyterlab    |         yielded = self.gen.throw(*exc_info)  # type: ignore
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/jupyterlab/handlers/extension_manager_handler.py",
> line 161, in _get_pkg_info
> jupyterlab    |         outdated = yield self._get_outdated()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 735, in run
> jupyterlab    |         value = future.result()
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 742, in run
> jupyterlab    |         yielded = self.gen.throw(*exc_info)  # type: ignore
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/jupyterlab/handlers/extension_manager_handler.py",
> line 195, in _load_outdated
> jupyterlab    |         app_options=self.app_options
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 735, in run
> jupyterlab    |         value = future.result()
> jupyterlab    |       File "/opt/conda/lib/python3.7/concurrent/futures/_base.py", line 425, in result
> jupyterlab    |         return self.__get_result()
> jupyterlab    |       File "/opt/conda/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
> jupyterlab    |         raise self._exception
> jupyterlab    |       File "/opt/conda/lib/python3.7/concurrent/futures/thread.py", line 57, in run
> jupyterlab    |         result = self.fn(*self.args, **self.kwargs)

> File "/opt/conda/lib/python3.7/site-packages/jupyterlab/commands.py", line 550, in get_latest_compatible_package_versions
> jupyterlab    |         return handler.latest_compatible_package_versions(names)
> jupyterlab    |       File "/opt/conda/lib/python3.7/site-packages/jupyterlab/commands.py", line 1675, in latest_compatible_package_versions
> jupyterlab    |         raise ValueError(msg % keys)
> jupyterlab    |     ValueError: "['@krassowski/jupyterlab-lsp@0.6.1', '@rahlir/theme-gruvbox@1.1.0', '@telamonian/theme-darcula@1.1.3', '@oriolmirosa/jupyterlab_materialdarker@0.4.3', 'jupyterlab-sos@0.5.3', 'transient-display-data@0.3.0', 'qgrid@1.1.1', 'jupyterlab-jupytext@1.0.2', '@jupyterlab/celltags@0.2.0', 'jupyterlab-chart-editor@0.2.0', 'jupyter-matplotlib@0.4.2', '@jupyter-widgets/jupyterlab-manager@1.1.0', 'ipysheet@0.4.3', 'jupyterlab-drawio@0.6.0', '@jupyterlab/toc@1.0.1', '@lckr/jupyterlab_variableinspector@0.3.0', '@mflevine/jupyterlab_html@0.1.4', '@agoose77/jupyterlab-markup@0.1.5', 'jupyterlab_autoversion@0.1.1', 'jupyterlab_conda@1.1.0', 'jupyterlab_toastify@2.3.2', 'jupyter-vuetify@1.1.1', 'jupyter-vue@1.0.0', '@ryantam626/jupyterlab_code_formatter@0.7.0', '@voziq/customgit@5.1.2', '@jupyterlab/git@0.8.2', 'nbdime-jupyterlab@1.0.0', '@jupyterlab/github@1.0.1', '@jupyter-voila/jupyterlab-preview@0.1.3']" is not a valid npm package

If I run vanilla Jupyterlab without (many) extensions installed, I get it working fine and am able to browse and install extensions throught Jupyterlab GUI.

**Comment by cbmiller2610** [comment]
Zireael, did you ever resolve this issue? I just started having the same issue. I can use commands like `jupyterlab labextension install jupyterlab-manager` to install packages, but then I see errors like `ValueError: "['jupyterlab-jupytext@1.1.1', '@jupyter-widgets/jupyterlab-manager@1.0.3', 'jupyterlab-plotly@4.6.0', 'plotlywidget@4.6.0']" is not a valid npm package `.

**Comment by jasongrout** [comment]
Often that error happens when there is some problem downloading an npm package.

However, your error seems to indicate that there is some problem specifying a package name. It seems to think that the *list* of package names is a single package that it should be trying to download.

What is the exact command you are running?

**Comment by cbmiller2610** [comment]
I would install all packages using `jupyter labextension install` (sorry, previous comment had a typo).  For example I was trying to reinstall all the necessary extensions for plotly and jupytext after having to reinstall node.js.  Then, when starting jupyter lab, I would see those errors in the log.  However, I think I "solved" it... and I'd guess it has something to do with the jupyterlab extensions manager.  I uninstalled all extensions using `jupyter labextensions uninstall`, and then did `jupyter lab clean`.  I then reinstalled all extensions, and did `jupyter lab build --minimize=False --dev-build=False`. Miraculously everything works and I don't see that error anymore.

**Comment by jasongrout** [comment]
@Zireael - is everything working for you now too?

**Comment by Overbryd** [comment]
@jasongrout I have the same problem building from a "vanilla" `jupyter/scipy-notebook:latest` (resolves to `8e3c844d40f5` 2020-09-10 at the time of writing).

<details>
  <summary>Here is my Dockerfile:</summary>

```
FROM jupyter/scipy-notebook:latest

# Interact with PostgreSQL data
RUN pip install \
  sqlalchemy \
  psycopg2-binary \
  ipython-sql \
  ipywidgets \
  streamz

# Fix nodejs/npm issues with @jupyter-widgets/jupyterlab-manager
# RUN conda install conda-forge nodejs

USER jovyan

# Jupyter real time stream processing
RUN jupyter nbextension enable --py widgetsnbextension \
  && jupyter labextension install \
    "@jupyter-widgets/jupyterlab-manager@2" \
    # Great plotting library that also works with streams
    "@bokeh/jupyter_bokeh"
```
</details>

Now when I start jupyterlab and load it in the browser, the following error message appears:

<details>
  <summary>Log output of the error "is not a valid npm package"</summary>

```console
jupyter_1  | [W 09:50:18.156 LabApp] Unhandled error
jupyter_1  | [E 09:50:18.156 LabApp] {
jupyter_1  |       "Host": "127.0.0.1:8888",
jupyter_1  |       "Connection": "keep-alive",
jupyter_1  |       "Pragma": "no-cache",
jupyter_1  |       "Cache-Control": "no-cache",
jupyter_1  |       "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36 OPR/70.0.3728.133",
jupyter_1  |       "X-Xsrftoken": "2|87457364|31599bc1985066fa9137a6eaab78c35b|1599504012",
jupyter_1  |       "Authorization": "token 452eb264b582c5ac26412bf6e958afbbcd615f6275446dbd",
jupyter_1  |       "Content-Type": "application/json",
jupyter_1  |       "Accept": "*/*",
jupyter_1  |       "Sec-Fetch-Site": "same-origin",
jupyter_1  |       "Sec-Fetch-Mode": "cors",
jupyter_1  |       "Sec-Fetch-Dest": "empty",
jupyter_1  |       "Referer": "http://127.0.0.1:8888/lab/workspaces/auto-Y?clone",
jupyter_1  |       "Accept-Encoding": "gzip, deflate, br",
jupyter_1  |       "Accept-Language": "en-GB,en-US;q=0.9,en;q=0.8",
jupyter_1  |       "Cookie": "_xsrf=2|87457364|31599bc1985066fa9137a6eaab78c35b|1599504012; username-127-0-0-1-8888=\"2|1:0|10:1599731381|23:username-127-0-0-1-8888|44:YWYxMTQ4YjgwYjk3NDRkYmE3OTg3ZmIyMmI4ZThkY2Y=|6f3fe3827d45093177d08bb5fedde3bd6a4b30214918671567be812b2a953b87\""
jupyter_1  |     }
jupyter_1  | [E 09:50:18.157 LabApp] 500 GET /lab/api/extensions?1599731381956 (172.20.0.1) 36011.75ms referer=http://127.0.0.1:8888/lab/workspaces/auto-Y?clone
jupyter_1  | [E 09:50:18.157 LabApp] Uncaught exception GET /lab/api/extensions?1599731382962 (172.20.0.1)
jupyter_1  |     HTTPServerRequest(protocol='http', host='127.0.0.1:8888', method='GET', uri='/lab/api/extensions?1599731382962', version='HTTP/1.1', remote_ip='172.20.0.1')
jupyter_1  |     Traceback (most recent call last):
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/web.py", line 1703, in _execute
jupyter_1  |         result = await result
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/web.py", line 1703, in _execute
jupyter_1  |         result = await result
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 742, in run
jupyter_1  |         yielded = self.gen.throw(*exc_info)  # type: ignore
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/handlers/extension_manager_handler.py", line 221, in get
jupyter_1  |         extensions = yield self.manager.list_extensions()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 735, in run
jupyter_1  |         value = future.result()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 742, in run
jupyter_1  |         yielded = self.gen.throw(*exc_info)  # type: ignore
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/handlers/extension_manager_handler.py", line 221, in get
jupyter_1  |         extensions = yield self.manager.list_extensions()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 735, in run
jupyter_1  |         value = future.result()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 742, in run
jupyter_1  |         yielded = self.gen.throw(*exc_info)  # type: ignore
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/handlers/extension_manager_handler.py", line 91, in list_extensions
jupyter_1  |         pkg_info = yield self._get_pkg_info(name, data)
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 735, in run
jupyter_1  |         value = future.result()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 742, in run
jupyter_1  |         yielded = self.gen.throw(*exc_info)  # type: ignore
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/handlers/extension_manager_handler.py", line 91, in list_extensions
jupyter_1  |         pkg_info = yield self._get_pkg_info(name, data)
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 735, in run
jupyter_1  |         value = future.result()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 742, in run
jupyter_1  |         yielded = self.gen.throw(*exc_info)  # type: ignore
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/handlers/extension_manager_handler.py", line 160, in _get_pkg_info
jupyter_1  |         outdated = yield self._get_outdated()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 735, in run
jupyter_1  |         value = future.result()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 742, in run
jupyter_1  |         yielded = self.gen.throw(*exc_info)  # type: ignore
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/handlers/extension_manager_handler.py", line 160, in _get_pkg_info
jupyter_1  |         outdated = yield self._get_outdated()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 735, in run
jupyter_1  |         value = future.result()
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 742, in run
jupyter_1  |         yielded = self.gen.throw(*exc_info)  # type: ignore
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/handlers/extension_manager_handler.py", line 191, in _load_outdated
jupyter_1  |         data = yield self.executor.submit(
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/tornado/gen.py", line 735, in run
jupyter_1  |         value = future.result()
jupyter_1  |       File "/opt/conda/lib/python3.8/concurrent/futures/_base.py", line 432, in result
jupyter_1  |         return self.__get_result()
jupyter_1  |       File "/opt/conda/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
jupyter_1  |         raise self._exception
jupyter_1  |       File "/opt/conda/lib/python3.8/concurrent/futures/thread.py", line 57, in run
jupyter_1  |         result = self.fn(*self.args, **self.kwargs)
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/commands.py", line 541, in get_latest_compatible_package_versions
jupyter_1  |         return handler.latest_compatible_package_versions(names)
jupyter_1  |       File "/opt/conda/lib/python3.8/site-packages/jupyterlab/commands.py", line 1682, in latest_compatible_package_versions
jupyter_1  |         raise ValueError(msg % keys)
jupyter_1  |     ValueError: "['jupyter-matplotlib@0.7.3', '@bokeh/jupyter_bokeh@2.0.3', '@jupyter-widgets/jupyterlab-manager@2.0.0']" is not a valid npm package
jupyter_1  | [W 09:50:18.158 LabApp] Unhandled error
jupyter_1  | [E 09:50:18.162 LabApp] {
jupyter_1  |       "Host": "127.0.0.1:8888",
jupyter_1  |       "Connection": "keep-alive",
jupyter_1  |       "Pragma": "no-cache",
jupyter_1  |       "Cache-Control": "no-cache",
jupyter_1  |       "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36 OPR/70.0.3728.133",
jupyter_1  |       "X-Xsrftoken": "2|87457364|31599bc1985066fa9137a6eaab78c35b|1599504012",
jupyter_1  |       "Authorization": "token 452eb264b582c5ac26412bf6e958afbbcd615f6275446dbd",
jupyter_1  |       "Content-Type": "application/json",
jupyter_1  |       "Accept": "*/*",
jupyter_1  |       "Sec-Fetch-Site": "same-origin",
jupyter_1  |       "Sec-Fetch-Mode": "cors",
jupyter_1  |       "Sec-Fetch-Dest": "empty",
jupyter_1  |       "Referer": "http://127.0.0.1:8888/lab/workspaces/auto-Y?clone",
jupyter_1  |       "Accept-Encoding": "gzip, deflate, br",
jupyter_1  |       "Accept-Language": "en-GB,en-US;q=0.9,en;q=0.8",
jupyter_1  |       "Cookie": "_xsrf=2|87457364|31599bc1985066fa9137a6eaab78c35b|1599504012; username-127-0-0-1-8888=\"2|1:0|10:1599731382|23:username-127-0-0-1-8888|44:MTlkNDYxN2NhNDUwNDdiYWFkYjFjNTMyMjZlZTk2Y2Q=|37cb2a155c2fefc6b4a8d9c74d8ae6afdf80e9c0569b7a96b3bb10471ccc44c8\""
jupyter_1  |     }
jupyter_1  | [E 09:50:18.162 LabApp] 500 GET /lab/api/extensions?1599731382962 (172.20.0.1) 35080.77ms referer=http://127.0.0.1:8888/lab/workspaces/auto-Y?clone
```
</details>

I have also tried the suggested workaround from the previous commenter.

<details>
  <summary>Here is the Dockerfile for the workaround</summary>

```
FROM jupyter/scipy-notebook:latest

# Interact with PostgreSQL data
RUN pip install \
  sqlalchemy \
  psycopg2-binary \
  ipython-sql \
  ipywidgets \
  streamz

# Fix nodejs/npm issues with @jupyter-widgets/jupyterlab-manager
# RUN conda install conda-forge nodejs

USER jovyan

# Jupyter real time stream processing
RUN jupyter nbextension enable --py widgetsnbextension \
  && jupyter labextension uninstall \
  && jupyter lab clean \
  && jupyter labextension install \
    "@jupyter-widgets/jupyterlab-manager@2" \
    # Great plotting library that also works with streams
    "@bokeh/jupyter_bokeh" \
  && juptyer lab build
```
</details>

Which also yields the same error as posted above.

**Comment by matt-erhart** [comment]
I'm also getting this error.

**Comment by CharlesARoy** [comment]
Just posting this here in case it can help anyone. 
I was also getting the "is not a valid npm package" for any Jupyter Lab extensions I tried installing from npm. 

Side note, from the [Jupyter Lab extensions documentation](https://jupyterlab.readthedocs.io/en/stable/user/extensions.html#installing-extensions), NPM packages are "source extension packages" that require Node.js. As of right now (2021-01-04), it seems that many of the most popular ones have not yet been adapted for Jupyter Lab 3.0.0 so you may want to install Jupyter Lab version 2.2.9 (e.g.:  `conda install -c conda-forge jupyterlab=2.2.9`)

Another side note, it seems that the nbextensions (such as [Spellchecker](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/spellchecker/README.html), [Execute Time](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/execute_time/readme.html), [Split Cells](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/splitcell/readme.html), and [Skip-Traceback](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/skip-traceback/readme.html)) are unofficial Jupyter Lab extensions and according to [their documentation](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/install.html):

> Most nbextensions here should work fine with jupyterhub... but won’t work with jupyterlab (because the jupyterlab javascript framework is different to notebook’s, and still rapidly changing under active development).


I read through #1982, #2943, #6845, #8982 and this issue (#7531) but none of them helped me solve my problem, though they _might_ help you. 

After reading what felt like every Google hit for "is not a valid npm package", I eventually realized (D'OH!) that the most helpful error codes to look up were the NPM ones listed in the logs (that look like): /var/folders/7m/mtl30ngj1r5_3lb1vg66qc5h0000gn/T/jupyterlab-debug-6u1qlrad.log

```
npm ERR! code ENOTFOUND
npm ERR! syscall getaddrinfo
npm ERR! errno ENOTFOUND
npm ERR! network request to https://registry.npmjs.org/jupyterlab-drawio failed, reason: getaddrinfo ENOTFOUND proxy.url.com
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
```
From the [NPM Login Errors section](https://docs.npmjs.com/common-errors#npm-login-errors) of the npm docs common errors, they suggested clearing the contents of the `~/.npmrc` file. After I did this, I continued getting the same "is not a valid npm package" error, but in the logs the errors changed to this:

```
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /Users/charles/.npm/_cacache/index-v5/74/f0
npm ERR! errno EACCES
npm ERR!
npm ERR! Your cache folder contains root-owned files, due to a bug in
npm ERR! previous versions of npm which has since been addressed.
npm ERR!
npm ERR! To permanently fix this problem, please run:
npm ERR!   sudo chown -R 501:20 "/Users/charles/.npm"
```
**TLDR**
After running `sudo chown -R 501:20 "/Users/charles/.npm"` I no longer got the "is not a valid npm package" error and instead got the much better "ValueError: The extension "jupyterlab-drawio" does not yet support the current version of JupyterLab." error, which finally meant that things were at least working properly. 

Ultimately, after reverting to Jupyter Lab version 2.2.9, I started getting a bunch of other errors. They persisted even after installing Jupyter Lab in its own environment, so I ultimately fully uninstalled and reinstalled Conda and after a clean install of everything, it all worked the first time without having to repeat any of the above. 

Hope this helps someone!

**Comment by jasongrout** [comment]
Thank you very much for posting your debug process and conclusion to help others!

**Comment by akitxu** [comment]
In my case I thought that the problem was because I was behind a corporate proxy, and finally I discovered that it was because I had not registered npm by doing

`npm config set registry http://registry.npmjs.org/

jlpm config set registry https://registry.yarnpkg.com/

**Comment by jasongrout** [comment]
Thanks for sharing as well!

Since we have several solutions here now for various configurations of the network that could generate this error, and no follow-up from the original submitter, I'll close this issue as resolved.
